### PR TITLE
Add Restore UUID Index Setting

### PR DESCRIPTION
**Pull request by original-brownbear**
Pre-requesite for #50278 to be able to uniquely identify index metadata by
its version fields and UUIDs when restoring into closed indices.

See related discussion here: https://github.com/elastic/elasticsearch/pull/50278#discussion_r382498267

### CODE REVIEW

**Review by ywelsch — CHANGES_REQUESTED** [review]
I wonder if we should generalize this to an `index.history.uuid` (which is also updated on dangling indices import, as well as whenever we run the unsafe bootstrap command). This makes this also more generally useful to characterize the identity of the index metadata.

**Comment by original-brownbear** [comment]
Jenkins run elasticsearch-ci/2 (random known aggs failure)

**Comment by original-brownbear** [comment]
Jenkins run elasticsearch-ci/bwc

File: x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/TransportResumeFollowAction.java [hunk 1]
```diff
@@ -442,6 +442,7 @@ static Settings filter(Settings originalSettings) {
         settings.remove(IndexSettings.INDEX_SOFT_DELETES_SETTING.getKey());
         settings.remove(IndexMetadata.SETTING_INDEX_VERSION_CREATED.getKey());
         settings.remove(IndexMetadata.SETTING_INDEX_UUID);
+        settings.remove(IndexMetadata.SETTING_RESTORE_UUID);
```
[the remark this answers is not in the document]
Comment (original-brownbear) [reply x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/TransportResumeFollowAction.java]: Jup makes sense to me => done.

**Comment by original-brownbear** [comment]
Thanks @ywelsch, I renamed the setting as suggested and added the functionality to dangling index import and unsafe bootstraps now.

**Review by ywelsch — APPROVED** [review]
LGTM

**Comment by original-brownbear** [comment]
Thanks Yannick!

**Pull request merged**